src/karmen_frontend/src/routes/organizations/add-printer.js

```diff
@@ -150,14 +163,19 @@ class AddPrinter extends React.Component {
                 )}
                 <FormInputs
                   definition={form}
+                  collapsibleDefinition={{
+                    definition: collapsibleForm,
+                    collapsedStateText: "Advanced options",
+                    expandedStateText: "Hide advanced options",
+                    updateValue: (name, value) => {
+                      this.setState({
+                        form: this.updateValue(collapsibleForm, name, value),
+                      });
+                    },
+                  }}
                   updateValue={(name, value) => {
                     this.setState({
-                      form: Object.assign({}, form, {
-                        [name]: Object.assign({}, form[name], {
-                          val: value,
-                          error: null,
-                        }),
-                      }),
+                      form: this.updateValue(form, name, value),
```

Comment:
The removed logic did much more than what is used now, are we sure about this?

Reply:
Are you sure? The deleted code has been extracted to a standalone method by copy paste with a minor modifications.

Reply:
OK, I didn't notice that. Closing.